app/models/lookup_value.rb

```diff
@@ -81,7 +81,8 @@ def ensure_fqdn_exists
     md = ensure_matcher(/fqdn=(.*)/)
     return md if md == true || md == false
     fqdn = md[1].split(LookupKey::KEY_DELM)[0]
-    return true if Host.unscoped.find_by_name(fqdn) || host_or_hostgroup.try(:new_record?) ||
+    fqdn_matched = Host.unscoped.where(:name => fqdn).any? || Host.unscoped.search_for("facts.fqdn=#{fqdn}").any?
```

Comment:
This is not really user-friendly, why not adding a new matcher 'shortname'? Maybe it'd be easy to add this matcher in another PR then use it here, just to not mix stuff and keep PRs simple.

Reply:
> This is not really user-friendly, why not adding a new matcher 'shortname'? Maybe it'd be easy to add this matcher in another PR then use it here, just to not mix stuff and keep PRs simple.


@dlobatog Ok let me see if I can state this in a clearer fashion. That line in lookup value is deceptively simple looking.

The bug is that Customer wants short hostnames when using the API But wants to have a fully formed fqdn in their puppet fact. They were able to get away with this setup before Host unification with Katello because Content Host and Host were 2 different things. However when they unified they wanted the host to have the short host name. They always wanted to keep the short host name but wanted the fqdn fact in puppet to reflect fully qualified name including the interface domain name.

Not just that. On top of this they want to be able Override certain puppet configuration (aka smart class parameters) based on the FQDN
Foreman code prior to this change made the assumption that Name = FQDN when matching which worked ok prior to shortname logic.

That is why we see code like

```return true if Host.unscoped.find_by_name(fqdn) || host_or_hostgroup.try(:new_record?) ||```

All my code does -> When Matching..
- Try matching by the existing method first. i.e See if you can find a host that has name given in the FQDN matcher . Identical to the line above
-  If that does not exist See if there is a host that has the FQDN puppet fact matching the value given in the FQDN matcher
that 2nd case is where this customer's matcher would match.

I would argue that this change improves the existing 'matching' setup by actually matching fqdns for real instead of assuming name given the foreman host == FQDN

Adding a shortname matcher is not really going to solve the issue for this customer because they want the existing FQDN matcher to actually match FQDN. Not name.

Reply:
If customer needs to combina short name and FQDN in single matcher, I think a new matcher should be added. Something like `fqdn_or_shortname`. We're trying to improve project usability, this would go the opposite direction. I think the best solution is to add `shortname` and `fqdn_or_shortname` to already existing `fqdn` and leave `fqdn` untouched.